### DISCUSSION TITLE
Problema improvviso con integrazione Netatmo

### DISCUSSION THREAD

**Discussion by Topolynx** [indomus/forum#278] [HUB: Home Assistant]
Buongiorno a tutti!
Vi chiedo una mano perché proprio non sto riuscendo a risolvere da me questo problema improvviso.
Stamattina, come ogni tanto capita, le mie valvole termostatiche Netatmo sono diventate irraggiungibili da HA. Ho riavviato il server come di consueto, cosa che risolve il problema momentaneo, ma stavolta non c'è stato verso di raggiungerle di nuovo. Dopo un po' di riavvii ho provato a disinstallare e reinstallare l'integrazione, ma quando sono andato ad effettuare l'autorizzazione (metodo yaml), la nuova pagina che si apre mi dà il seguente errore:
_Missing code or state parameter in https://XXXXXX.duckdns.org:8196/auth/external/callback?error=invalid_client&state=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJmbG93X2lkIjoiOTYyNzkwMjcyOGMyNDAyOWE0NjRmNzU4ZGM2YjBjZDIiLCJyZWRpcmVjdF91cmkiOiJodHRwczovL3ZhZmZhbmd1bGUuZHVja2Rucy5vcmc6ODE5Ni9hdXRoL2V4dGVybmFsL2NhbGxiYWNrIn0.epJxeihyt99ByYc6EyPwnW26DWBlq5qZ-xaYt0dnWRA_.
Non so che cosa fare più, ho anche provato ad andare sul sito developer di netatmo, creare un'applicazione impostando il redirect URI col mio indirizzo https://XXXXXX.duckdns.org:8196 ed aggiornare username e password nel configuration.yaml con le id e secret collegate al progetto ma niente.
Avete qualche suggerimento?

**Comment by Scialla** [comment]
Ci sono problemi con l'integrazione Netatmo, soprattutto con l'ultimo aggiornamento di HA il 2021.5.0 (edit: in realtà non dipende dalla versione ma dalle API che sono probabilmente cambiate)
L'integrazione è rotta per quanto riguarda le valvole termostatiche ma continua a funzionare per il meteo e le cam.
Ho aperto una Issue su github, puoi seguirla da qui:
https://github.com/home-assistant/core/issues/50203

**Reply by Topolynx** [reply]
Oh my god ti ringrazio @Scialla, ci stavo a diventare pazzo... <3

**Reply by Scialla** [reply]
eh, ci ho litigato pure io ieri sera :)

**Reply by indomus** [reply]
raga una ricerchina su github di home assistant fatela sempre...
https://github.com/home-assistant/core/issues?q=is%3Aissue+is%3Aopen+netatmo

**Reply by Scialla** [reply]
> raga una ricerchina su github di home assistant fatela sempre...
> https://github.com/home-assistant/core/issues?q=is%3Aissue+is%3Aopen+netatmo

L'ho aperta io la Issue :) ( e l'ho anche linkata sopra)

**Reply by indomus** [reply]
sei troppo forte

**Reply by snakuzzo** [reply]
il più forte, direi

**Reply by Scialla** [reply]
non vi dico cosa siete voi due :D :D :D

**Comment by essegi65** [comment]
Ho appena fatto l'aggiornamento a 2015.5.1 e anche io ho notato che l'attributo del livello batteria termostato netatmo è sparito:
hvac_modes:

auto
heat
'off'
min_temp: 7
max_temp: 35
target_temp_step: 0.5
preset_modes:
away
boost
Frost Guard
Schedule
current_temperature: 21.1
temperature: 16.5
hvac_action: idle
preset_mode: Schedule
friendly_name: Netatmo Soggiorno
supported_features: 17

Ho visto la issue su github ma sbaglio o ancora non c'è la soluzione?

**Reply by Scialla** [reply]
Ho appena aperto una nuova Issue, come richiesto da cgtobi
https://github.com/home-assistant/core/issues/50320

**Reply by essegi65** [reply]
scusami @Scialla ma ho visto che la issue del tuo link è chiusa, quindi il problema è risolto? se si in che modo visto che nella issue non mi sembra riportata la soluzione (il mio inglese fa schifo forse non ho letto bene).

**Reply by Scialla** [reply]
L'ho chiusa io perché ne era già stata aperta un'altra ed era un doppione. Comunque ci sta già lavorando e ci sarà soluzione a breve.
Poi ho aperto altra issue per il problema delle impostazioni delle valvole

**Reply by essegi65** [reply]
ok, grazie

**Comment by essegi65** [comment]
In attesa della soluzione relativa al problema del sensore della batteria, ho notato che il termostato netatmo supporta 17 funzionalità.
Qualcuno riesce a dirmi dove posso trovare una descrizione, o elenco, di queste funzionalità?
"supported_features: 17"
grazie

**Reply by Scialla** [reply]
Hanno sistemato il problema dell'attributo ma, ora, non è più numerico ma è una stringa, quindi bisogna rifare i sensori...
